docs/core/compatibility/aspnet-core/7.0/environment-variable-precedence.md

```diff
@@ -0,0 +1,42 @@
+---
+title: "Breaking change: ASPNET-prefixed environment variable precedence"
+description: Learn about the breaking change in ASP.NET Core 7.0 where ASPNET-prefixed environment variables now have the lowest precedence for WebApplicationBuilder.
+ms.date: 12/02/2022
+---
+# ASPNET-prefixed environment variable precedence
+
+Starting in .NET 7, and only when using the `WebApplicationBuilder` host, command-line arguments and `DOTNET_`-prefixed environment variables override `ASPNET_`-prefixed environment variables when reading from default host configuration sources. These sources are used to read host variables, such as the content root path and environment name, when the `WebApplicationBuilder` is constructed and serve as a base for app configuration.
+
+`ASPNET_`-prefixed environment variables now have the lowest precedence of all of the default host configuration sources for `WebApplicationBuilder`. For other hosts, such as `ConfigureWebHostDefaults` and `WebHost.CreateDefaultBuilder`, `ASPNET_`-prefixed environment variables still have the highest precedence.
+
+## Version introduced
+
+ASP.NET Core 7.0
+
+## Previous behavior
+
+`ASPNET_`-prefixed environment variables overrode command-line arguments and `DOTNET_`-prefixed environment variables.
```

Comment:
```suggestion
`ASPNET_`-prefixed environment variables overrode command-line arguments and `DOTNET_`-prefixed environment variables when configuring `WebApplicationBuilder`'s [default host configuration](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-7.0#default-host-configuration-sources).
```

This is not true generally. Command-line variables still override `ASPNET_`-prefixed variables given the generic host. We hope to fix this for the generic host too with a seperate breaking change in .NET 8. https://github.com/dotnet/aspnetcore/issues/17963#issuecomment-1320631406

And we always have overriden `ASPNET_` environment variables for application configuration. This breaking change is specific to reading from default host configuration sources.